src/pkg/services/m365/m365.go

```diff
@@ -61,12 +65,87 @@ func UsersCompat(ctx context.Context, acct account.Account) ([]*User, error) {
 func UsersCompatNoInfo(ctx context.Context, acct account.Account) ([]*UserNoInfo, error) {
 	errs := fault.New(true)
 
-	users, err := usersNoInfo(ctx, acct, errs)
+	usersResult, err := usersNoInfo(ctx, acct, errs)
 	if err != nil {
 		return nil, err
 	}
 
-	return users, errs.Failure()
+	return usersResult, errs.Failure()
+}
+
+// UserHasMailbox returns true if the user has any mailbox enabled
+// false otherwise, and a nil pointer and an error in case of error
+func UserHasMailbox(ctx context.Context, acct account.Account, userID string) (*bool, error) {
+	uapi, err := makeUserAPI(acct)
+	result := false
+
+	if err != nil {
+		return nil, clues.Wrap(err, "getting mailbox").WithClues(ctx)
+	}
+
+	requestParameters := users.ItemMailFoldersRequestBuilderGetQueryParameters{
+		Select: []string{"id"},
+		Top:    ptr.To[int32](1), // if we get any folders, then we have access.
+	}
+
+	options := users.ItemMailFoldersRequestBuilderGetRequestConfiguration{
+		QueryParameters: &requestParameters,
+	}
+
+	_, err = uapi.GetMailFolders(ctx, userID, options)
+	if err != nil {
+		if graph.IsErrUserNotFound(err) {
+			logger.CtxErr(ctx, err).Error("user not found")
+			return nil, graph.Stack(ctx, clues.Stack(graph.ErrResourceOwnerNotFound, err))
+		}
+
+		if !graph.IsErrExchangeMailFolderNotFound(err) ||
+			clues.HasLabel(err, graph.LabelStatus(http.StatusNotFound)) {
+			logger.CtxErr(ctx, err).Error("getting user's mail folder")
+			return nil, err
+		}
+
+		logger.Ctx(ctx).Info("resource owner does not have a mailbox enabled")
+
+		return &result, nil
+	}
+
+	result = true
+
+	return &result, nil
```

Comment:
Once the return value removes the pointer, this can flatten to a direct value return.
```suggestion
	return true, nil
```